### DISCUSSION TITLE
Schema `pre` and `post` using `arguments`

### DISCUSSION THREAD

**Discussion by hasezoey** [Automattic/mongoose#12075] [Q&A]
Is there currently a specific reason on why `Schema.prototype.pre` and `Schema.prototype.post` use `arguments` instead of explicit parameters like other functions?

https://github.com/Automattic/mongoose/blob/3445d240886e508962d7dd6a8481715a3cb373da/lib/schema.js#L1588-L1607

https://github.com/Automattic/mongoose/blob/3445d240886e508962d7dd6a8481715a3cb373da/lib/schema.js#L1646-L1665

**Comment by vkarpov15** [comment]
No. The arguments are listed here: https://github.com/vkarpov15/kareem/blob/db2dcc714f8158e50a81c2b53f47e87a06dd6772/index.js#L435 so feel free to create a PR to add these params.
